drivers/spi/spi_ll_stm32.c

```diff
@@ -414,6 +414,16 @@ static int transceive(struct device *dev,
 
 	spi_context_release(&data->ctx, ret);
 
+#ifdef CONFIG_SPI_SLAVE
+	/*
+	 * in spi slave case (ret > 0) may represent the number of frames
+	 * received and not the I/O status. So, we should discriminate this.
+	 */
+	if (spi_context_is_slave(&data->ctx) && (spi_stm32_get_err(spi) == 0)) {
+		return ret; /* number of frames received */
+	}
+#endif
+
```

Comment:
Actually, I would have rewritten how ret is handled here in transceive: removing the below test for logging error, and removing the ret ? -EIO : 0; to me, in case of error, ret should already have the right error code.

Reply:
you'll need to make the function returning errors giving directly the right error code with relevant changes depending on whether you are slave or master.

Reply:
I don't understand why this should be Slave specific, from API point of view Slave or Master transfers should be the same, with the same error return scheme.
ret != 0 means an error, ret = 0 mean you no error
why do you want the number of frames when an error occurs ?

Reply:
This is part of the SPI changes that happened recently. In slave mode: if an error occurs it will need to return a negative error code, but in case of success it will have to return the number of received frames.
Master mode is as before: 0 on success, an negative error code otherwise.

Reply:
As I commented few minutes ago after re-pushing the commit, the situation is now clean and aligned with other SPI controllers (like dw or intel), except when CONFIG_SPI_STM32_INTERRUPT is disabled. In that case I had to add a check that in all other cases is already done in the context part. 
Question: do we really need the stm32 case with interrupt disabled?

Reply:
My bad, I missed the point, sorry